weave/trait/pairing/basic_open_close_installation_state_trait.proto

```diff
@@ -0,0 +1,116 @@
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file specifies the basic open close installation state trait.
+ */
+
+syntax = "proto3";
+
+package weave.trait.pairing;
+
+import "wdl/wdl_options.proto";
+
+option java_outer_classname = "WeaveInternalBasicOpenCloseInstallationStateTrait";
+option objc_class_prefix = "SCM";
+
+/**
+ *  @brief
+ *    Basic Open Close Installation State Trait
+ *
+ *  A Weave Common service published trait that provides a means for an
+ *  application to set a an open close sensor device to different modes
+ *  during installation, and to query the current installation mode of
+ *  the device.
+ *
+ *  The consumer of this information is any actor driving pairing flows
+ *  for open close sensor devices.
+ */
+message BasicOpenCloseInstallationStateTrait {
+  option (wdl.message_type) = TRAIT;
+
+  option (wdl.trait) = {
+    stability: ALPHA,
+    vendor_id: 0x0000,
+    id:        0x1402,
+    version:   1
+  };
+
+  //  @brief
+  //    Modes during installation
+  //
+  //  Different user interface modes used during device installation
+  //
+  enum InstallationMode {
+    // Unspecified status, default/initial value of this enum
+    INSTALLATION_MODE_UNSPECIFIED = 0;
+
+    // Idle: the device provides no active UX when in this state.
+    INSTALLATION_MODE_IDLE = 1;
+
+    // Placement guidance: the device provides visual or other indications,
+    // such as lighting up, to assist the user with placement of the device
+    INSTALLATION_MODE_PLACEMENT_GUIDANCE = 2;
+
+    // Open close test: the device provides visual or other indications
+    // to assist the user with an open close test, and optionally,
+    // performs analysis/calibration during such test
+    INSTALLATION_MODE_OPEN_CLOSE_TEST = 3;
+  }
+
+  //  @brief
+  //    Open close test result Status
+  //
+  //  The current result of an open close test.
+  //
+  enum OpenCloseTestResult {
+    OPEN_CLOSE_TEST_RESULT_UNSPECIFIED = 0;
+    OPEN_CLOSE_TEST_RESULT_UNKNOWN = 1;
+    OPEN_CLOSE_TEST_RESULT_INVALID = 2;
+    OPEN_CLOSE_TEST_RESULT_VALID = 3;
+  }
+
+  //  @brief
+  //    Response to command to set installation mode.
+  enum SetInstallationModeResponseType {
+    SET_INSTALLATION_MODE_RESPONSE_TYPE_UNSPECIFIED = 0;
+    SET_INSTALLATION_MODE_RESPONSE_TYPE_SUCCESS = 1;
+    SET_INSTALLATION_MODE_RESPONSE_TYPE_ERROR = 2;
+  }
+
+  // ---------- PROPERTIES ---------- //
+  option (wdl.properties) = { writable: READ_ONLY };
+  
+  // Which mode the device is currently on.
+  InstallationMode installation_mode = 1;
+
+  OpenCloseTestResult self_test_result = 2;
```

Comment:
Is this necessary?

Reply:
I think it's good to give the option of being able to convey a result for the test. It may not be used in all cases, but I can see potential usefulness for it.